sale_blanket_order/models/blanket_orders.py

```diff
@@ -301,7 +301,10 @@ def action_confirm(self):
             sequence_obj = self.env["ir.sequence"]
             if order.company_id:
                 sequence_obj = sequence_obj.with_company(order.company_id.id)
-            name = sequence_obj.next_by_code("sale.blanket.order")
+            if order.name == "Draft":
+                name = sequence_obj.next_by_code("sale.blanket.order")
+            else:
+                name = order.name
             order.write({"confirmed": True, "name": name})
```

Comment:
It is only a small issue, but wouldn't it be better not to write the name at all, if it does not change?

vals = {"confirmed": True}
if order.name == "Draft":
    vals["name"] = sequence_obj.next_by_code("sale.blanket.order")

order.write(vals)